tests/unix_stream.rs

```diff
@@ -451,6 +451,8 @@ where
 
     assert!(stream.take_error().unwrap().is_none());
 
+    assert_would_block(stream.read(&mut buf));
```

Comment:
This is the same behaviour that tcp_stream has and is functionally important for the edge triggered emulation to work.  Basically if you only read exactly the data you need then nothing would trigger the EAGAIN case and the fd wouldn't be re-registered with POLLIN.  Technically this doesn't feel like a big issue in practice though since in any real world application you'd have to read all available buffered data up to EAGAIN to know for sure that subsequent calls to epoll would actually tell you if data was ready.  There could be contrived and very bizarre usages though that would break...